client/src/scene/SceneGameArena.ts

```diff
@@ -206,46 +226,48 @@ export class SceneGameArena extends Phaser.Scene {
     }
 
     private updateFalling(scene: SceneGameArena) {
-        // fall the tetromino
-        // if (can fall)
-        //    fall
-        // else
-        //    TODO place on board
-
         // NOTE: other players' tetrominoes are treated as static blocks, although they are synced shortly before this function
 
         const state = scene.gameState;
         const board = state.board;
         const tetro = state.currentTetromino;
 
-        if (this.canTetroFall(tetro, board)) {
-            tetro.position[0] += 1;
-
+        if (
+            tetro.moveIfCan(board, (tetro) => {
+                tetro.position[0] += 1;
+                if (tetro.position[0] > BOARD_SIZE) {
+                    tetro.position[0] = 0;
+                }
```

Comment:
If this is just a temporary handling of this case, a comment should be added.

Reply:
this check might be better put into moveIfCan, so deleting it for now.